plans/TOOLING.md

```diff
@@ -0,0 +1,241 @@
+
+### Milestones
+
+#### :arrow_right: Create a Parser/Validator
+The first tool we would have to implement is the AsyncAPI parser/validator. The parser/validator is a core piece of all the other tooling we'll build in the future. Therefore it's a top priority for us.
+
+##### Requirements:
+1. It MUST support YAML and JSON as input formats.
+2. It MUST support the following schema formats: OpenAPI schemas, JSON Schema Draft 04-07, Protobuf, and Avro.
+3. It MUST support plugging in a new format parser.
+4. It MUST be implemented as a Go package and [exported as a C shared object](https://medium.com/learning-the-go-programming-language/calling-go-functions-from-other-languages-4c7d8bcc69bf).
+5. It MUST have wrappers in the following languages: Java and Node.js.
+
+#### :arrow_right: Create code generation framework
+1. Design a code generation solution.
+2. Keep in mind how to make the code generation solution suitable for documentation generation too. Maybe by improving [asyncapi/generator](https://github.com/asyncapi/generator)?
+3. Implement code generation templates for the following languages: Java, Node.js, and Go.
+
+##### Dependencies:
+* Parser/Validator
+
+#### :arrow_right: SDKs
+Investigate, define requirements, and implement, at least, two SDKs: one for Java <sup>1</sup> and another for Node.js. Ideally, we should find a way to automate the process of creating new SDKs, or to make it as fast and smooth as possible.
+
+##### Suggestions:
+* Implement an SDK in Go.
+* Create wrappers for other languages.
+
+##### Dependencies:
+* Parser/Validator
+
+#### :arrow_right: Implement code generation templates using SDKs
+Implement templates in our code generation tool for a simple Java application and a simple Node.js application, both using our SDKs.
+
+##### Dependencies:
+* Parser/Validator
+* SDKs
+
+#### :arrow_right: Add support for AsyncAPI 2.0.0 in AsyncAPI React component
+_Requirements to be discussed with [@derberg](https://github.com/derberg) and the Kyma/SAP team._
+
+##### Suggestions:
+* Extract parser implementation to another package.
+* Make sure all parser dependencies are compatible with a browser.
+
+##### Dependencies:
+* Parser/Validator
+
+#### :arrow_right: Add support for AsyncAPI 2.0.0 in all [asyncapi/generator](https://github.com/asyncapi/generator) templates
+So far, only HTML and Markdown templates will require work.
+
+##### Dependencies:
+* _(none)_
+
+#### :arrow_right: Improve online code editor (editor.asyncapi.org)
+Most probably this will consist in ditching the current editor in favor of the playground [@derberg](https://github.com/derberg) and the Kyma/SAP team have created.
```

Comment:
Is there a link to the 'playground'? Does it allow creation / modification of JSON/YAML AsyncAPI definitions like the current editor?

Reply:
here you go https://www.asyncapi.com/asyncapi-react/

Reply:
@MikeRalphson I repeat here the same question but for you :) What is your skill level with Rust??

Reply:
Total beginner. I made a few hacks to the Google Gnostic project, so I've actually done more with Go so far, but I like the *sound* of Rust.

Reply:
@derberg Let's discuss the requirements later. I just wanted to communicate that it's on the roadmap.